src/NServiceBus.Core/ObjectBuilder/ContainerComponent.cs

```diff
@@ -0,0 +1,124 @@
+﻿namespace NServiceBus
+{
+    using System;
+    using System.Collections.Generic;
+    using Container;
+    using ObjectBuilder;
+    using ObjectBuilder.Common;
+    using Settings;
+
+    class ContainerComponent
+    {
+        public ContainerComponent(SettingsHolder settings)
+        {
+            this.settings = settings;
+        }
+
+        public IConfigureComponents ContainerConfiguration { get; private set; }
+
+        public IBuilder Builder { get; private set; }
+
+        public void UseContainer<T>(Action<ContainerCustomizations> customizations = null) where T : ContainerDefinition, new()
+        {
+            customizations?.Invoke(new ContainerCustomizations(settings));
+
+            UseContainer(typeof(T));
+        }
+
+        public void UseContainer(Type definitionType)
+        {
+            UseContainer(definitionType.Construct<ContainerDefinition>().CreateContainer(settings));
+        }
+
+        public void UseContainer(IContainer container)
+        {
+            customContainer = container;
+        }
+
+        public void UseExternallyManagedBuilder(IBuilder builder)
+        {
+            ownsContainer = false;
```

Comment:
This should be in `InitializeWithExternallyManagedContainer` instead, right?

Reply:
Yea, that would be the correct place 👍